postgres-native-sqldelight-driver/src/commonTest/kotlin/app/softwork/sqldelight/postgresdriver/PostgresNativeDriverTest.kt

```diff
@@ -135,4 +147,70 @@ class PostgresNativeDriverTest {
         val results = driver.copy("1\n2\n")
         assertEquals(2, results)
     }
+
+    @Ignore
+    @Test
+    fun listenerTest() = runTest(dispatchTimeoutMs = 10.seconds.inWholeMilliseconds) {
+        val other = backgroundScope.PostgresNativeDriver(
+            host = "localhost",
+            port = 5432,
+            user = "postgres",
+            database = "postgres",
+            password = "password"
+        )
+
+        val driver = backgroundScope.PostgresNativeDriver(
+            host = "localhost",
+            port = 5432,
+            user = "postgres",
+            database = "postgres",
+            password = "password"
+        )
+
+        val results = MutableStateFlow(0)
+        val listener = object : Query.Listener {
+            override fun queryResultsChanged() {
+                results.update { it + 1 }
+            }
+        }
+        driver.addListener(listener, arrayOf("foo", "bar"))
+        println("AFTER adding")
+        withContext(Dispatchers.Default) {
+            val setupSocket = 1.seconds
+            println("BEFORE DELAY")
+            delay(setupSocket)
+            println("AFTER DELAY")
+        }
+        other.notifyListeners(arrayOf("foo"))
+
+        println("AFTER DELAY II")
+        other.notifyListeners(arrayOf("foo", "bar"))
+        other.notifyListeners(arrayOf("foo"))
+        //driver.removeListener(listener, arrayOf("foo"))
+        other.notifyListeners(arrayOf("foo"))
+        assertEquals(5, results.value)
+    }
+
+    @Test
+    fun listenerTest2() = runTest(dispatchTimeoutMs = 10.seconds.inWholeMilliseconds) {
+        val listener = flow {
+            while (true) {
+                suspendCoroutine<Unit> {
+                    select(1, null, null, null, null)
+                }
```

Comment:
Won't this hang forever? The `Continuation<Unit>` called with `resumeWith`.

Reply:
Yes. Biggest problem is `select` that blocks the `coroutineScope`. This was only a test to investigate this behavior.